ports/esp32/network_ppp.c

```diff
@@ -115,61 +115,124 @@ static void pppos_client_task(void *self_in) {
     vTaskDelete(NULL);
 }
 
-STATIC mp_obj_t ppp_active(size_t n_args, const mp_obj_t *args) {
-    ppp_if_obj_t *self = MP_OBJ_TO_PTR(args[0]);
-
-    if (n_args > 1) {
-        if (mp_obj_is_true(args[1])) {
-            if (self->active) {
-                return mp_const_true;
-            }
+STATIC mp_obj_t ppp_connect_py(size_t n_args, const mp_obj_t *args, mp_map_t *kw_args) {
+    enum { ARG_self, ARG_auth_method, ARG_user_name, ARG_password };
+    static const mp_arg_t allowed_args[] = {
+        { MP_QSTR_self, MP_ARG_OBJ | MP_ARG_REQUIRED, {.u_obj = MP_OBJ_NULL} },
```

Comment:
self is not needed in this arg array

Reply:
Removed `self` from the arg array and uses arg[0] instead.